influxdb/client.py

```diff
@@ -284,15 +285,18 @@ def write(self, data, params=None, expected_response_code=204,
         headers = self._headers
         headers['Content-type'] = 'application/octet-stream'
 
-        if params:
+        if params and 'precision' in params:
             precision = params.get('precision')
         else:
             precision = None
 
         if protocol == 'json':
             data = make_lines(data, precision).encode('utf-8')
         elif protocol == 'line':
-            data = ('\n'.join(data) + '\n').encode('utf-8')
+            if type(data) == list:
```

Comment:
I am not super-experienced Pythonist, however I think that this is not the best solution. Or is it?
Since sequence is requested, user may provide both list or tuple. In this case if tuple is found it will be not casted to list but put into list. Then the following join will fail because instead of str it will find tuple. Cast by list() would be much better, because str will be put into list, and tuple converted into list. Win-win I guess. If any other data structure was provided - exception is justified.